text/0000-inferred-type-aliases.md

```diff
@@ -0,0 +1,696 @@
+For example, you will be allowed to write:
+
+```rust
+use std::ops::Range;
+
+type MyIterator = _;
+
+fn one_to_hundred() -> MyIterator {
+    let range: Range<u8> = 0..100;
+    range
+}
+
+fn main() {
+    // OK! because this is transparent inference.
+    let _numbers: Range<u8> = one_to_hundred();
+}
+```
+
+You can also optionally enforce that `MyIterator` must at least be an `Iterator`:
+
+```rust
+type MyIterator: Iterator<Item = u8> = _;
+```
+
+This works the same way in associated types:
+
+```rust
+// Let's pretend this type is more interesting in actuality.
+#[derive(Debug)]
+struct SomeImportantStuff;
+struct MyType;
+trait MyTrait {
+    type MyImplementationDetail;
+    fn my_public_api(&self) -> Self::MyImplementationDetail;    
+}
+
+// Our implementation:
+impl MyTrait for MyType {
+    // Here we employ type inference to our aid:
+    type MyImplementationDetail = _;
+
+    fn my_public_api(&self) -> Self::MyImplementationDetail {
+        // Here goes logic that determines the
+        // actual type of `MyImplementationDetail`...
+        SomeImportantStuff
+    }
+}
+
+fn main() {
+    // OK! Inferred type is transparent.
+    let _detail: SomeImportantStuff = MyType.my_public_api();
+}
+```
+
+Same as with type aliases, you may also enforce a trait bound:
+
+```rust
+    // Still transparent and will leak everything about the type:
+    type MyImplementationDetail: Debug = _;
+```
+
+Another way to think of the feature proposed in this RFC is that it is the
+`_` inside `type Alias = _;` is the [transparent] version of the [opaque]
+and [inferred] `type Alias = impl Trait` ([RFC 2515]) or equivalently
+`existential type Alias: Trait;` ([RFC 2071]).
+
+## Dos and Don’ts
+[do_dont]: #dos-and-don'ts
+
+1. **Do not** use `_` inside type aliases as a means of *abstraction*.
+  Aliases that use `_` inside them will *leak* all information about the
+  exact type that is inferred. Changes to the capabilities / affordances /
+  operations offered by the underlying inferred type will cause breakage in
+  consumers of your APIs that depend on those affordances. Therefore, be extra
+  careful when using `pub` on `type Alias = _;` or when using `_` for an
+  associated type of an implementation for a public trait + public type.
+
+2. **Do** use `_` as a means of eliding complex implementation details that
+  are not important to the users of your API.
+  This applies in particular to custom derive macros for traits with
+  associated types.
+
+3. **Do**, assuming that you keep in mind 1), use `_` as a means to name the
+   type of an expression with a [voldemort type] somewhere inside it.
+
+## Teaching this
+
+The teaching of the [transparent] and [inferred] types can be explained in
+conjunction with explaining the usual `_` feature in other places.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Grammar
+
+The grammar of `type` aliases and associated `type`s are extended to permit
```

Comment:
This seems like a comparatively large large grammar extension for very little benefit. You gain only a little extra documentation, without even being able to document nested occurrences `_` (which seems like a natural future extension), at the cost of a modification to the grammar.

Reply:
Provided justification in the rationale now :)